source/slang/type-layout.cpp

```diff
@@ -1599,6 +1604,28 @@ SimpleLayoutInfo GetLayoutImpl(
 
             return info;
         }
+        else if (auto globalGenParam = declRef.As<GlobalGenericParamDecl>())
+        {
+            SimpleLayoutInfo info;
+            info.alignment = 0;
+            info.size = 0;
+            info.kind = LayoutResourceKind::GenericResource;
+            if (outTypeLayout)
+            {
+                auto genParamTypeLayout = new GenericParamTypeLayout();
+                genParamTypeLayout->decl = globalGenParam.getDecl();
+                // we should have already populated ProgramLayout::genericEntryPointParams list at this point,
+                // so we can find the index of this generic param decl in the list
+                genParamTypeLayout->paramIndex = findGenericParam(context.targetReq->layout->genericEntryPointParams, genParamTypeLayout->decl);
+                genParamTypeLayout->type = type;
+                genParamTypeLayout->rules = rules;
+                genParamTypeLayout->findOrAddResourceInfo(LayoutResourceKind::GenericResource)->count++;
+                genParamTypeLayout->uniformAlignment = 0;
+                genParamTypeLayout->addResourceUsage(LayoutResourceKind::Uniform, 0);
+                *outTypeLayout = genParamTypeLayout;
+            }
```

Comment:
Is there a reason why you are adding zero bytes of `Uniform` data here? What happens if you drop that line? I would like it is types don't report resources that they don't actually use, just to keep the information as clean as possible.

Also: I can see how this resource usage will propagate up through aggregate types, but I'm not as clear about:

- Will this resource usage properly "infect" later fields in a structure, so that we properly recognize that they can't be given useful offsets?

- Will this new resource usage propagate up through containers like constant buffers, so that it will always be exposed at the global scope?